src/lib/y2storage/disk_size.rb

```diff
@@ -609,10 +609,48 @@ def floor(unit_size)
     # @example
     #   x = DiskSize.KB(1)   #=> <DiskSize 0.98 KiB (1000)>
     #   x.to_human_string    #=> "0.98 KiB"
-    def to_human_string
+    def to_human_string(round_method = :round)
       return "unlimited" if unlimited?
-      size, unit = human_string_components
-      format("%.2f %s", size, unit)
+      float, unit_s = human_string_components
+      rounded = (float * 100).public_send(round_method) / 100.0
+      # A plain "#{rounded} #{unit_s}" would not keep trailing zeros
+      format("%.2f %s", rounded, unit_s)
+    end
+
+    # A human readable string that does not exceed the exact size.
+    #
+    # If we have 4.999 GiB of space, and ask the user how much of that
+    # should be used, prefilling the "Size" widget with the maximum
+    # rounded up to "5.00 GiB" it will then fail validation.
```

Comment:
Ok. If the user has an option to not waste space by selecting "maximum size", then is fair enough (I assume there is no rounding at all in that case).

As a completely unrelated note, libstorage-ng does not use cylinders at all (that are an obsolete unit). I guess we can still "simulate" them in the "custom region" part. Or maybe we should switch to sectors or anything that makes sense nowadays.